### PR TITLE
Add codeowners file

### PR DESCRIPTION
**Pull request by twsearle**
## Description

The github [CODEOWNERS file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) allows repository owners to specify owners of a repository or particular directories within a repo. This flexible approach allows automatic notification/addition of reviewers to pull requests relevant for that team or individual that is responsible for the repository. Later, we could add restrictions so that reviewers listed would have to accept the change before it could be merged, but I am not suggesting that right now.

Advantages:
 * guarantee that one of the reviewers has the ability to merge, meaning that we won't have any time between a change being marked accepted and merged.
 * the author of the PR would not have to figure out whose approval they need to get a change merged.

## Issue(s) addressed

Experiment to  implement suggestion in https://github.com/orgs/JCSDA-internal/discussions/144

### CODE REVIEW

**Review by matthewrmshin — APPROVED** [review]
Perhaps consider adding a deputy as well?

**Comment by twsearle** [comment]
> Perhaps consider adding a deputy as well?

I can add you if you like, but at the moment this codeowners thing just makes sure I am added by default to every PR - very light touch. I don't really want to spam multiple people with emails when there is no need? I won't be required to accept the PR for it to be merged, so doesn't matter if I get hit by a bus! :) I agree though we would want something else if we require a review from a particular group. I am really only adding this as a demo - where I think it will be most useful is in ufo - we can say, add met office team as a codeowners for a bunch of met office checks just so they get automatically notified or add a global owner to make sure one person has a handle on merging the changes in at JCSDA. I see every PR come in on this repo anyway so I don't think this will make much difference... just playing around.

**Comment by matthewrmshin** [comment]
@twsearle Understood. All good as it is. 😄

**Pull request merged**
